omod/src/main/java/org/openmrs/module/mksreports/web/controller/MKSReportsManageController.java

```diff
@@ -0,0 +1,150 @@
+/**
+ * The contents of this file are subject to the OpenMRS Public License
+ * Version 1.0 (the "License"); you may not use this file except in
+ * compliance with the License. You may obtain a copy of the License at
+ * http://license.openmrs.org
+ *
+ * Software distributed under the License is distributed on an "AS IS"
+ * basis, WITHOUT WARRANTY OF ANY KIND, either express or implied. See the
+ * License for the specific language governing rights and limitations
+ * under the License.
+ *
+ * Copyright (C) OpenMRS, LLC.  All Rights Reserved.
+ */
+package org.openmrs.module.mksreports.web.controller;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.HashMap;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.xml.transform.Result;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.sax.SAXResult;
+import javax.xml.transform.stream.StreamSource;
+
+import org.apache.fop.apps.FOUserAgent;
+import org.apache.fop.apps.Fop;
+import org.apache.fop.apps.FopFactory;
+import org.apache.fop.apps.MimeConstants;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.mksreports.reports.PatientHistoryReportManager;
+import org.openmrs.module.patientsummary.PatientSummaryResult;
+import org.openmrs.module.patientsummary.PatientSummaryTemplate;
+import org.openmrs.module.patientsummary.api.PatientSummaryService;
+import org.openmrs.module.reporting.report.ReportDesign;
+import org.openmrs.module.reporting.report.service.ReportService;
+import org.openmrs.util.OpenmrsClassLoader;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.ModelMap;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+
+/**
+ * The main controller.
+ */
+@Controller
+public class MKSReportsManageController {
+	
+	@RequestMapping(value = "/module/mksreports/manage", method = RequestMethod.GET)
+	public void manage(ModelMap model) {
+		model.addAttribute("user", Context.getAuthenticatedUser());
+	}
+	
+	/**
+	 * The path to the style sheet for Patient History reports.
+	 */
+	public static final String PATIENT_HISTORY_XSL_PATH = "patientHistoryFopStylesheet.xsl";
+	
+	/**
+	 * Receives requests to run a patient summary.
+	 * 
+	 * @param patientId the id of patient whose summary you wish to view
+	 * @param summaryId the id of the patientsummary you wish to view
+	 */
+	@RequestMapping(value = "/module/mksreports/patientHistory")
```

Comment:
We need this URI to protected through authorization, either through some annotation that might do it out of the box (such as what Spring Security would do, but I doubt it), or manually by checking the user's privileges and returning an error 403 if unauthorized.

Reply:
@mks-d , I added two strategies to protect the report, one is adding privilege in mksreports_extension.json, the other is  forcing to check the privilege in the render  Context.requirePrivilege(MksReportPrivilegeConstants.VIEW_PATIENT_HISTORY);